packages/amplify-cli/src/lib/init-steps/preInitSetup.js

```diff
@@ -0,0 +1,87 @@
+const { execSync } = require('child_process');
+const { getPackageManager } = require('../packageManagerHelpers');
+const { normalizePackageManagerForOS } = require('../packageManagerHelpers');
+const { generateLocalEnvInfoFile } = require('./s9-onSuccess');
+const url = require('url');
+
+async function run(context) {
+  if (context.parameters.options.app) {
+    // Setting up a sample app
+    context.print.warning('Note: Amplify does not have knowledge of the url provided');
+    const repoUrl = context.parameters.options.app;
+    await validateGithubRepo(context, repoUrl);
+    await cloneRepo(context, repoUrl);
+    await installPackage();
+    await setLocalEnvDefaults(context);
+  }
+  return context;
+}
+
+/**
+ * Checks whether a url is a valid remote github repository
+ *
+ * @param repoUrl the url to validated
+ * @throws error if url is not a valid remote github url
+ */
+async function validateGithubRepo(context, repoUrl) {
+  try {
+    url.parse(repoUrl);
+    execSync(`git ls-remote ${repoUrl}`, { stdio: 'ignore' });
+  } catch (e) {
+    context.print.error('Invalid remote github url');
+    process.exit(1);
+  }
+}
+
+/**
+ * Clones repo from url to current directory (must be empty)
+ *
+ * @param repoUrl the url to be cloned
+ */
+async function cloneRepo(context, repoUrl) {
+  try {
+    execSync(`git clone ${repoUrl} .`, { stdio: 'inherit' });
+  } catch (e) {
```

Comment:
Is this the only error that could happen when cloning the repo? I can foresee the there could be other errors, due to network issues. The reason I wanted you to ensure directory was empty was then we specifically know `git clone` is going to fail. 

May be not printing anything would also work as we are `execSync` is inheriting the stdio

Reply:
Thats a good point. I'll make a check for an empty directory